docs/transpile/create-a-transpiler-plugin.ipynb

```diff
@@ -0,0 +1,311 @@
+    "- `options`, a free-form configuration dictionary for plugin-specific options. If your plugin has these configuration options you\n",
+    "should clearly document them.\n",
+    "\n",
+    "The `run` method returns a [QuantumCircuit](/api/qiskit/qiskit.circuit.QuantumCircuit)\n",
+    "representing the circuit synthesized from that high-level object.\n",
+    "It is also allowed to return `None`, indicating that the plugin is unable to synthesize the given high-level object.\n",
+    "The actual synthesis of high-level objects is performed by the\n",
+    "[HighLevelSynthesis](/api/qiskit/qiskit.transpiler.passes.HighLevelSynthesis)\n",
+    "transpiler pass.\n",
+    "\n",
+    "In addition to the `run` method, there are a number of property methods that need to be defined.\n",
+    "See [HighLevelSynthesisPlugin](/api/qiskit/qiskit.transpiler.passes.synthesis.plugin.HighLevelSynthesisPlugin) for documentation of all required properties.\n",
+    "\n",
+    "Let's define our HighLevelSynthesisPlugin subclass:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "3c0e59d4-85b0-4157-824b-b1f6220e83ad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.synthesis import synth_clifford_bm\n",
+    "from qiskit.transpiler.passes.synthesis.plugin import HighLevelSynthesisPlugin\n",
+    "\n",
+    "\n",
+    "class MyCliffordSynthesisPlugin(HighLevelSynthesisPlugin):\n",
+    "    def run(\n",
+    "        self, high_level_object, coupling_map=None, target=None, qubits=None, **options\n",
+    "    ) -> QuantumCircuit:\n",
+    "        if high_level_object.num_qubits <= 3:\n",
+    "            return synth_clifford_bm(high_level_object)\n",
+    "        else:\n",
+    "            return None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1285dbfd-04ce-494b-8a01-a58040035dbd",
+   "metadata": {},
+   "source": [
+    "This plugin synthesizes objects of type [Clifford](/api/qiskit/qiskit.quantum_info.Clifford) that have\n",
+    "at most 3 qubits, using the `synth_clifford_bm` method.\n",
+    "\n",
+    "Now, we expose the plugin by adding an `entry-points` table in `pyproject.toml` under the `qiskit.synthesis` namespace.\n",
+    "To expose this plugin in a package called `my_qiskit_plugin`, we would add the following in `pyproject.toml`:\n",
+    "\n",
+    "```toml\n",
+    "[project.entry-points.\"qiskit.synthesis\"]\n",
+    "\"clifford.special\" = \"my_qiskit_plugin.module.plugin:MyCliffordSynthesisPlugin\"\n",
+    "```\n",
+    "\n",
+    "The `name` consists of two parts separated by a dot (`.`):\n",
+    "- The name of the type of [Operation](/api/qiskit/qiskit.circuit.Operation) that the plugin synthesizes (in this case, `clifford`). Note that this string corresponds to the [`name`](/api/qiskit/qiskit.circuit.Operation#name) attribute of the Operation class, and not the name of the class itself.\n",
+    "- The name of the plugin (in this case, `special`)."
```

Comment:
Would be nice to round this off by explaining how they can check it's registered correctly. I.e. installing then running

```python
>>> from qiskit.transpiler.passes.synthesis import high_level_synthesis_plugin_names
>>> high_level_synthesis_plugin_names("clifford")
['ag', 'bm', 'default', 'greedy', 'layers', 'lnn', 'special']
```

Same for the other two examples

Reply:
Thanks! Good call with the `my_` prefixes, makes it a lot clearer what is plugin-specific rather than a qiskit keyword